Firestore/core/src/remote/remote_store.cc

```diff
@@ -528,16 +539,22 @@ absl::optional<TargetData> RemoteStore::GetTargetDataForTarget(
                                         : absl::optional<TargetData>{};
 }
 
+void RemoteStore::RestartNetwork() {
+  is_network_enabled_ = false;
+  DisableNetworkInternal();
+  online_state_tracker_.UpdateState(OnlineState::Unknown);
+  write_stream_->InhibitBackoff();
+  watch_stream_->InhibitBackoff();
+  EnableNetwork();
+}
+
 void RemoteStore::HandleCredentialChange() {
   if (CanUseNetwork()) {
     // Tear down and re-create our network streams. This will ensure we get a
     // fresh auth token for the new user and re-fill the write pipeline with new
     // mutations from the `LocalStore` (since mutations are per-user).
     LOG_DEBUG("RemoteStore %s restarting streams for new credential", this);
-    is_network_enabled_ = false;
-    DisableNetworkInternal();
-    online_state_tracker_.UpdateState(OnlineState::Unknown);
-    EnableNetwork();
+    RestartNetwork();
```

Comment:
Looks like this PR could also make reconnection faster upon credentials' change? If correct, perhaps it's worth mentioning in the PR's description?

Reply:
Only if the credential change happened when while offline, which is probably uncommon (since Auth needs a network connection as well).